src/basic/proc-cmdline.c

```diff
@@ -402,7 +402,9 @@ int proc_cmdline_get_bool(const char *key, ProcCmdlineFlags flags, bool *ret) {
         if (r < 0)
                 return r;
         if (r == 0) { /* key not specified at all */
-                *ret = false;
+                if (!skip_missing) /* else the original value of ret is retained */
+                        *ret = false;
+
                 return 0;
```

Comment:
Hmm, if `skip_missing` is true, then `ret` is not initialized...
How about to rename the flag, say, `missing_to_false` or so, and
```c
if (r == 0) {
        if (!missing_to_false)
                return -ENOKEY;

        *ret = false
        return 0;
}
```

Reply:
Moreover, introducing a new flag, say, `PROC_CMDLINE_MISSING_TO_FALSE` is better, than introducing a boolean parameter.

Reply:
Or, to preserve most of existing call, conversely `PROC_CMDLINE_MISSING_TO_FATAL` may be better.